simplified-ui-accounts/src/main/java/org/nypl/simplified/ui/accounts/AccountListRegistryViewModel.kt

```diff
@@ -0,0 +1,116 @@
+package org.nypl.simplified.ui.accounts
+
+import androidx.lifecycle.ViewModel
+import hu.akarnokd.rxjava2.subjects.UnicastWorkSubject
+import io.reactivex.android.schedulers.AndroidSchedulers
+import io.reactivex.disposables.CompositeDisposable
+import org.librarysimplified.services.api.Services
+import org.nypl.simplified.accounts.api.AccountEvent
+import org.nypl.simplified.accounts.api.AccountProviderDescription
+import org.nypl.simplified.accounts.registry.api.AccountProviderRegistryEvent
+import org.nypl.simplified.accounts.registry.api.AccountProviderRegistryStatus
+import org.nypl.simplified.accounts.registry.api.AccountProviderRegistryType
+import org.nypl.simplified.buildconfig.api.BuildConfigurationServiceType
+import org.nypl.simplified.profiles.controller.api.ProfilesControllerType
+import org.nypl.simplified.threads.NamedThreadPools
+import org.slf4j.LoggerFactory
+import java.net.URI
+
+class AccountListRegistryViewModel : ViewModel() {
+
+  private val services =
+    Services.serviceDirectory()
+
+  private val accountRegistry =
+    services.requireService(AccountProviderRegistryType::class.java)
+
+  private val profilesController =
+    services.requireService(ProfilesControllerType::class.java)
+
+  private val buildConfig =
+    services.requireService(BuildConfigurationServiceType::class.java)
+
+  private val backgroundExecutor =
+    NamedThreadPools.namedThreadPool(1, "simplified-registry-io", 19)
+
+  private val logger =
+    LoggerFactory.getLogger(AccountListRegistryViewModel::class.java)
+
+  private val subscriptions: CompositeDisposable =
+    CompositeDisposable(
+      this.accountRegistry.events
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(this::onAccountRegistryEvent),
+      this.profilesController.accountEvents()
+        .observeOn(AndroidSchedulers.mainThread())
+        .subscribe(this::onAccountEvent)
+    )
+
+  private fun onAccountRegistryEvent(event: AccountProviderRegistryEvent) {
+    this.accountRegistryEvents.onNext(event)
+  }
+
+  private fun onAccountEvent(event: AccountEvent) {
+    this.accountCreationEvents.onNext(event)
+  }
+
+  override fun onCleared() {
+    super.onCleared()
+    this.subscriptions.clear()
+    this.backgroundExecutor.shutdown()
```

Comment:
:+1: 

So many people forget to shut down exectors. They extend `AutoCloseable` in modern Java, but ... you know... Android...